src/main/java/edu/harvard/iq/dataverse/api/Files.java

```diff
@@ -317,6 +322,64 @@ public Response replaceFileInDataset(
         }
             
     } // end: replaceFileInDataset
+
+    /**
+     * Delete an Existing File 
+     * 
+     * @param id file ID or peristent ID
+     */
+    @DELETE
+    @Path("{id}")
+    public Response deleteFileInDataset(@PathParam("id") String fileIdOrPersistentId){
+
+        if (!systemConfig.isHTTPUpload()) {
+            return error(Response.Status.SERVICE_UNAVAILABLE, BundleUtil.getStringFromBundle("file.api.httpDisabled"));
+        }
+        // (1) Get the user from the API key
+        User authUser;
+        try {
+            authUser = findUserOrDie();
+        } catch (AbstractApiBean.WrappedResponse ex) {
+            return error(Response.Status.FORBIDDEN, BundleUtil.getStringFromBundle("file.addreplace.error.auth"));
+        }
+
+        msg("DELETE!");
+
+        // (2) Delete
+        boolean deletePhysicalFile = false;
+        try {
+            DataFile dataFile = findDataFileOrDie(fileIdOrPersistentId);
+            List<FileMetadata> fileToDelete = dataFile.getFileMetadatas();
+            Dataset dataset = dataFile.getOwner();
+            DatasetVersion v = dataset.getOrCreateEditVersion();
+            DataverseRequest dvRequest2 = createDataverseRequest(authUser);
+            deletePhysicalFile = !dataFile.isReleased();
+
+            UpdateDatasetVersionCommand update_cmd = new UpdateDatasetVersionCommand(dataset, dvRequest2,  fileToDelete, v);
+            update_cmd.setValidateLenient(true);
+
+            try {
+                commandEngine.submit(update_cmd);
+            } catch (CommandException ex) {
+                return error(BAD_REQUEST, "Delete failed for file ID: " + fileIdOrPersistentId + ": " + ex.getMessage());
+            } catch (EJBException ex) {
+                return error(BAD_REQUEST, "Delete failed for file ID: " + fileIdOrPersistentId + ": " + ex.getMessage());
+            }
+    
+            if (deletePhysicalFile) {
+                try {
+                    fileService.finalizeFileDelete(dataFile.getId(), fileService.getPhysicalFileToDelete(dataFile));
+                } catch (IOException ioex) {
+                    logger.warning("Failed to delete the physical file associated with the deleted datafile id="
+                            + dataFile.getId() + ", storage location: " + fileService.getPhysicalFileToDelete(dataFile));
+                }
+            }
+        } catch (WrappedResponse ex) {
+            return error(BAD_REQUEST, "There was no file found for ID: " + fileIdOrPersistentId);
+        }
+
+        return ok(deletePhysicalFile);
+    }
```

Comment:
I'm just putting this on the last line because this is where tests go.

Can we please have some integration tests to ensure this new functionality is working?